src/types/token_amount.ts

```diff
@@ -0,0 +1,54 @@
+// Utils
+import { BigNumber } from "../../utils/bignumber";
+
+// Types
+import { Token } from "../types";
+
+export class TokenAmount {
+    public static fromRawAmount(rawAmount: BigNumber, symbol: string): TokenAmount {
+        return new TokenAmount(rawAmount, new Token(symbol));
+    }
+
+    public static fromDecimalAmount(decimalAmount: BigNumber, symbol: string) {
+        const token = new Token(symbol);
+        const rawAmount = TokenAmount.convertToRaw(decimalAmount, token.numDecimals);
+
+        return new TokenAmount(rawAmount, token);
+    }
+
+    private static convertToRaw(decimalAmount: BigNumber, numDecimals: BigNumber): BigNumber {
+        return decimalAmount.mul(new BigNumber(10).pow(numDecimals.toNumber()));
+    }
+
+    private static convertToDecimal(rawAmount: BigNumber, numDecimals: BigNumber): BigNumber {
+        return rawAmount.div(new BigNumber(10).pow(numDecimals.toNumber()));
+    }
+
+    public readonly rawAmount: BigNumber;
+    private token: Token;
+
+    private constructor(rawAmount: BigNumber, token: Token) {
```

Comment:
Are we sure that this `private` keyword will have any effect in when this is transpiled down to JS?  I suspect users will still be able to initialize via the constructor

Reply:
This has been addressed by passing in a union type to the constructor and removing the private keyword.

Reply:
see: https://github.com/dharmaprotocol/dharma.js/pull/139